Common/ConfigLoader.cs

```diff
@@ -76,9 +75,9 @@ protected static ConfigClass LoadFromFile(string fileName)
                 return config;
             }
 
-            var serializer = new XmlSerializer(typeof(ConfigClass));
             using (var reader = new StreamReader(FileName))
             {
+                var serializer = new XmlSerializer(typeof(ConfigClass));
                 return (ConfigClass)serializer.Deserialize(reader);

```

Comment:
this cast could throw, maybe this should be handled?

Reply:
you are right, the `reader` could throw as well, but:
what if you are reading a file with invalid contents? either the `serializer` will throw when trying to deserialize it or it will accept the invalid content silently (not sure about the `XmlSerializer` class) and the [cast will throw ](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions#type-conversion-exceptions-at-run-time). either way it never hurts to be extra careful when doing casts ...